### PR TITLE
Jb52049

### PR DESCRIPTION
**Pull request by martyone**

### CODE REVIEW

**Comment by martyone** [comment]
Just the very last commit belongs to this PR

File: sdk-setup/src/mb2 [hunk 1]
```diff
@@ -3838,8 +3885,28 @@ done
...
+        mkdir -p "$STATEDIR" || exit
+        echo '*' > "$STATEDIR/.gitignore"
+        ;;
+    check | run | ssh )
+        # These can deal with STATEDIR missing
+        ;;
+    *)
+        if [[ ! -d $STATEDIR ]]; then
+            if ancestor=$(find_upwards "$PWD" ".$WE"); then
+                fatal "The command needs to be used from the top of the build tree ($ancestor)"
```
Comment (vigejolla) [inline sdk-setup/src/mb2:3902]: I was expecting to see automatically changing pwd to top of build tree, but this is actually even better, as there is less room for mistakes :)

**Review by vigejolla — APPROVED** [review]
LGTM

**Comment by martyone** [comment]
Fixed self-test ^

**Comment by vigejolla** [comment]
still LGTM

**Pull request merged**